src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
-        return psiMethod.body != null
-    }
+class JavaPsiClassWrapper(private val psiClass: PsiClass) : PsiClassWrapper {
+    override val name: String get() = psiClass.name ?: ""
 
-    /**
-     * Checks if the method is a default method of an interface.
-     *
-     * @param psiMethod the PSI method of interest
-     * @return true if the method is a default method of an interface, false otherwise
-     */
-    private fun isMethodDefault(psiMethod: PsiMethod): Boolean {
-        if (!isMethodConcrete(psiMethod)) return false
-        return psiMethod.containingClass?.isInterface ?: return false
-    }
+    override val methods: List<PsiMethodWrapper> get() = psiClass.methods.map { JavaPsiMethodWrapper(it) }
 
-    /**
-     * Checks if a PSI method is a default constructor.
-     *
-     * @param psiMethod the PSI method of interest
-     * @return true if the PSI method is a default constructor, false otherwise
-     */
-    private fun isDefaultConstructor(psiMethod: PsiMethod): Boolean {
-        return psiMethod.isConstructor && psiMethod.body?.isEmpty ?: false
-    }
+    override val allMethods: List<PsiMethodWrapper> get() = psiClass.allMethods.map { JavaPsiMethodWrapper(it) }
 
-    /**
-     * Checks if a PSI class is an abstract class.
-     *
-     * @param psiClass the PSI class of interest
-     * @return true if the PSI class is an abstract class, false otherwise
-     */
-    private fun isAbstractClass(psiClass: PsiClass): Boolean {
-        if (psiClass.isInterface) return false
+    override val qualifiedName: String get() = psiClass.qualifiedName ?: ""
+
+    override val text: String get() = psiClass.text
 
-        val methods = PsiTreeUtil.findChildrenOfType(psiClass, PsiMethod::class.java)
-        for (psiMethod: PsiMethod in methods) {
-            if (!isMethodConcrete(psiMethod)) {
-                return true
+    override val fullText: String
+        get() {
+            var fullText = ""
+            val fileText = psiClass.containingFile.text
+
+            // get package
+            packagePattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n\n"
+            }
+
+            // get imports
+            importPattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n"
             }
+
+            // Add class code
+            fullText += psiClass.text
+
+            return fullText
         }
 
-        // check if a class is noted as abstract in the text
-        return psiClass.text.replace(" ", "")
-            .contains("abstractclass${psiClass.name}", ignoreCase = true)
+    override val superClass: PsiClassWrapper? get() = psiClass.superClass?.let { JavaPsiClassWrapper(it) }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
```

Comment:
Properties and methods are overridden in a chaotic order; it is better to move all properties to the top of the class, leaving methods below.